### PR TITLE
issue reviews done

### PR DESCRIPTION
**Pull request by umayangag**
reviews changes for issue #646

### CODE REVIEW

File: results-tabulation-api/util/__init__.py [hunk 1]
```diff
@@ -158,12 +158,12 @@ def get_sum_of_all_and_nan_otherwise(array):
     return result
 
 
-def input_is_valid(content_array):
+def validate_tally_sheet_version_request_content_special_characters(content_array):
     invalid_strings = ["'", "\"", "<", ">", "=", ",", ";"]
     for array_item in content_array:
-        for value in array_item:
-            text_value = str(array_item[value])
+        if "strValue" in array_item:
```
Comment (dinukadesilva) [inline results-tabulation-api/util/__init__.py]: Have to check if "strValue" is not None

File: results-tabulation-api/api/TallySheetVersionApi/__init__.py [hunk 2]
```diff
@@ -168,9 +168,10 @@ def create(tallySheetId, body):
     tally_sheet = TallySheet.get_by_id(tallySheetId=tallySheetId)
 
     # validate user inputs to prevent XSS attacks
-    if not input_is_valid(request_body.get("content")):
-        raise InvalidInputException(
-            message="Invalid input detected. Use of disallowed characters/invalid input length detected",
+    input_is_valid, error_message = validate_tally_sheet_version_request_content_special_characters(request_body.get("content"))
+    if not input_is_valid:
```
Comment (dinukadesilva) [inline results-tabulation-api/api/TallySheetVersionApi/__init__.py]: Move the exception to the function itself.

**Review by dinukadesilva — APPROVED** [review]

**Review by dinukadesilva — CHANGES_REQUESTED** [review]
Few changes are requestd

File: results-tabulation-api/util/__init__.py (hunk 1 above)
Comment (umayangag) [reply results-tabulation-api/util/__init__.py]: done

File: results-tabulation-api/api/TallySheetVersionApi/__init__.py (hunk 2 above)
Comment (umayangag) [reply results-tabulation-api/api/TallySheetVersionApi/__init__.py]: done

File: results-tabulation-api/util/__init__.py (hunk 1 above)
Comment (dinukadesilva) [reply results-tabulation-api/util/__init__.py]: You could combine two if clauses to one

**Review by dinukadesilva — CHANGES_REQUESTED** [review]
One minor thing

**Review by dinukadesilva — APPROVED** [review]

**Pull request merged**